templates/vue/src/store/actions.js

```diff
@@ -71,6 +80,13 @@ export async function updateUserProgress() {
 }
 
 export async function completeNode({ commit, dispatch, getters }, nodeId) {
+  if (!wpData.wpUserId) {
+    const progressObj = JSON.parse(localStorage.getItem(LOCAL_PROGRESS_ID))
+    const nodeProgress = progressObj[nodeId] || {}
+    nodeProgress.completed = true
+    localStorage.setItem(LOCAL_PROGRESS_ID, JSON.stringify(progressObj))
+  }
+
   await client.completeNode(nodeId)
   commit("updateNode", {
```

Comment:
Small suggestion: Shouldn't we only make this API call if the user is logged in? Seeing as we know it's not doing anything.

Reply:
Good point, I notice there are a few other API calls that might be redundant as well. Thanks for pointing it out!

Reply:
Good point, I notice there are a few other API calls that might be redundant as well. Thanks for pointing it out!